### PR TITLE
feat: allow to use .PreviousTag on templates

### PR DESCRIPTION
**Pull request by caarlos0**
Previous tag will be available as a template field. It will be empty if there are no previous tag (e.g. first ever release of a repo).

Changelog still works by using the first commit on those cases (as it already does today).

### CODE REVIEW

**Comment by caarlos0** [comment]
one problem with this: on the first tag, this will actually be a commit... 🤔 which per-se is not that bad, its just that the name will not convey what it actually is at some point 🤔

File: internal/pipe/git/git.go [hunk 1]
```diff
@@ -203,6 +211,14 @@ func getTag() (string, error) {
 	return tag, err
 }
 
+func getPreviousTag(current string) (string, error) {
+	if tag := os.Getenv("GORELEASER_PREVIOUS_TAG"); tag != "" {
```
Comment (blacktop) [inline internal/pipe/git/git.go:215]: :+1:

**Comment by blacktop** [comment]
> one problem with this: on the first tag, this will actually be a commit... 🤔 which per-se is not that bad, its just that the name will not convey what it actually is at some point 🤔

Maybe a default `v0.0.0` would make sense?  I tried `v0.0.0` and a commit hash in github's URL and both 404ed (I know that is not the only purpose of this feature, however.  I kinda feel like a zero-ed out version would make more sense than a commit hash?

**Comment by blacktop** [comment]
I'm not intimately familiar with your code, but everything looked good to me. 👍

**Comment by caarlos0** [comment]
> Maybe a default `v0.0.0` would make sense? I tried `v0.0.0` and a commit hash in github's URL and both 404ed (I know that is not the only purpose of this feature, however. I kinda feel like a zero-ed out version would make more sense than a commit hash?

my comment was outdated, it will now be empty as stated in the docs... which makes sense, as if there is no previous tag, that's what I would expect...

**Review by dirien — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**